django/thunderstore/repository/api/experimental/tests/test_api_experimental.py

```diff
@@ -19,3 +30,51 @@ def test_api_experimental(client, active_package_listing, community_site):
         result[0]["package"]["full_name"]
         == active_package_listing.package.full_package_name
     )
+
+
+@pytest.mark.django_db
+def test_api_experimental_upload_package(
+    api_client,
+    user,
+    manifest_v1_data,
+    package_category,
+    uploader_identity,
+):
+    icon_raw = io.BytesIO()
+    icon = Image.new("RGB", (256, 256), "#FF0000")
+    icon.save(icon_raw, format="PNG")
+
+    readme = "# Test readme".encode("utf-8")
+    manifest = json.dumps(manifest_v1_data).encode("utf-8")
+
+    files = [
+        ("README.md", readme),
+        ("icon.png", icon_raw.getvalue()),
+        ("manifest.json", manifest),
+    ]
+
+    zip_raw = io.BytesIO()
+    with ZipFile(zip_raw, "a", ZIP_DEFLATED, False) as zip_file:
+        for name, data in files:
+            zip_file.writestr(name, data)
+
+    UploaderIdentityMember.objects.create(
+        user=user,
+        identity=uploader_identity,
+        role=UploaderIdentityMemberRole.owner,
+    )
+
+    api_client.force_authenticate(user=user)
+    response = api_client.post(
+        "/api/experimental/package/version/",
+        {
+            "author_name": uploader_identity.name,
+            "categories": json.dumps([package_category.pk]),
+            "has_nsfw_content": True,
+            "file": SimpleUploadedFile("mod.zip", zip_raw.getvalue()),
+        },
+        HTTP_ACCEPT="application/json",
+    )
+    assert response.status_code == 200
+    response = response.json()
+    assert PackageVersion.objects.filter(pk=response["pk"]).exists()
```

Comment:
Since we don't want to nor need to expose the PK outside, you should check this based on package + author name and package version. There's the `thunderstore.repository.package_reference.PackageReference` class for this that you can use to do e.g. `PackageReference.parse("package-reference_here-1.0.0").exists()`